service/src/lib.rs

```diff
@@ -141,8 +141,8 @@ pub fn new_full(config: Configuration<CustomConfiguration, GenesisConfig>)
 {
 	use substrate_network::DhtEvent;
 
-	let is_authority = config.roles.is_authority();
 	let is_collator = config.custom.collating_for.is_some();
+	let is_authority = config.roles.is_authority() && !is_collator;
```

Comment:
I don't know if this breaks some assumption in the code though

Reply:
What kind of assumptions? As collator we are currently a full node and this line expresses this.

Reply:
ok I should have put this comment on the if is_collator then return.

maybe there is more logic wrong there, like network_gossip register two times is this correct ?

https://github.com/paritytech/polkadot/blob/4a983a8bd812bd2120eb31897a94ddffc090cd39/service/src/lib.rs#L208-L211

https://github.com/paritytech/polkadot/blob/4a983a8bd812bd2120eb31897a94ddffc090cd39/service/src/lib.rs#L280

Reply:
This is the stuff I need :) Otherwise the node always disconnects, because of unknown gossip messages.

Reply:
I still don't understand why in case of a collator we register two times: one time as validator and another time as non_authority_validator

Reply:
This is not only done for collators, also for full nodes. 
I think this probably can be moved into the if.

Reply:
@andresilva confirmed that this is a bug, as you highlighted. We will open another pr to fix this.